lib/features/steps/multipart_request_steps.rb

```diff
@@ -0,0 +1,130 @@
+require 'test/unit'
+require 'open-uri'
+require 'json'
+require 'cgi'
+
+include Test::Unit::Assertions
+
+# @!group Multipart request assertion steps
+
+# Verifies a request contains the correct Content-Type header and some contents
+# for a multipart/form-data request
+#
+# @param request [Hash] The payload to test
+def valid_multipart_form_data?(request)
+  content_regex = Regexp.new('^multipart\\/form-data; boundary=[\\h-]+$')
+  content_header = request[:request]['Content-Type']
+  assert_match(content_regex, content_header)
+  assert(request[:body].size.positive?, "Multipart request payload contained #{request[:body].size} fields")
+end
+
+# Verifies that the request contains multipart form-data
+Then('the request is valid multipart form-data') do
+  valid_multipart_form_data?(Server.current_request)
+end
+
+# Verifies all received requests contain multipart form-data
+Then('all requests are valid multipart form-data') do
+  Server.stored_requests.all? { |request| valid_multipart_form_data?(request) }
+end
+
+# Tests the number of fields a multipart request contains.
+#
+# @step_input part_count [Integer] The number of expected fields
+Then('the multipart request has {int} fields') do |part_count|
+  parts = Server.current_request[:body]
+  assert_equal(part_count, parts.size)
+end
+
+# Tests the multipart request has at least one field.
+Then('the multipart request has a non-empty body') do
+  parts = Server.current_request[:body]
+  assert(parts.size.positive?, "Multipart request payload contained #{parts.size} fields")
+end
+
+# (Deprecated) Retained for backwards compatibility.
+# Use "the payload field {string} is not null"
```

Comment:
YARD has an `@deprecated` tag that should put this in the generated docs so it's more obvious

Reply:
Ok, we'll seek to update this in a future PR.